cmd/tailscale/cli/up.go

```diff
@@ -543,7 +543,12 @@ func runUp(ctx context.Context, cmd string, args []string, upArgs upArgsT) (retE
 
 	watchCtx, cancelWatch := context.WithCancel(ctx)
 	defer cancelWatch()
-	watcher, err := localClient.WatchIPNBus(watchCtx, 0)
+	watcher, err := localClient.WatchIPNBus(watchCtx, ipn.NotifyInitialState)
+	if err != nil {
+		return err
+	}
+	// Read the initial state and throw it away, to make sure the watcher is watching before we call Start().
+	_, err = watcher.Next()
```

Comment:
This is at least a succinct definition of the problem, but the intent of the API was that this was _already_ the case, so I'd prefer to fix the API somehow than require all callers to cargo-cult this.